ios/Sources/MaaS/UI/Components/Badge.swift

```diff
@@ -1,65 +1,309 @@
 import SwiftUI
 import Swappable
+import MaasTheme
 
+///  A swappable Badge component
+/// - Parameters:
+/// <-->
 public struct Badge: View, Swappable {
 
+    public enum BadgeType {
+        case small
+        case medium
+    }
+
     public struct InputType {
-        let color: Color?
+        let type: BadgeType
+        let backgroundColor: Color?
+        let foregroundColor: Color?
         let icon: UIImage?
         let text: String?
+        let alternativeBadgesColors: [Color]
+        let subBadge: UIImage?
```

Comment:
@stonevyt called this `subbadgeIcon` in Compose, would be great to strive for consistency 🌟

Reply:
Yes! missed this one